relayer/chains/wasm/provider.go

```diff
@@ -70,10 +70,10 @@ func (p *Provider) Init(ctx context.Context, homePath string, kms kms.KMS) error
 
 // Wallet returns the wallet of the provider
 func (p *Provider) Wallet() sdkTypes.AccAddress {
-	if p.wallet == nil {
-		ctx := context.Background()
-		done := p.SetSDKContext()
-		defer done()
+	ctx := context.Background()
+	done := p.SetSDKContext()
+	defer done()
+	if p.wallet == nil || p.wallet.GetAddress().Empty() {
```

Comment:
any reason for pulling the block outside for all the cases instead of where the wallet is nil or wallet address is empty similar to earlier implementation ?

Reply:
we have to use correct sdk context whenever we need wallets, default is cosmos.

The cosmos chains have their own prefixes and this may be also the reason behind the bug. Settings its own context is sufficient but we may never know so we'll handle the edge case by checking the empty address.